Sources/Growth/AppReviewManager.swift

```diff
@@ -146,6 +156,30 @@ public class AppReviewManager: ObservableObject {
     return mainCriteriaSatisfied && subCriteriaSatisfied
   }
   
+  public func shouldShowNewsRatingCard() -> Bool {
+    // Check if main and sub criteria is satisfied for new rating card presentation
+    guard checkLogicCriteriaSatisfied(for: .newsRatingCard) else {
+      return false
+    }
+    
+    // Check at least minDaysBetweenRatingCardPresented days passed since last card presentation
+    var daysSinceLastRequest = 0
+    if let previousRequest = Preferences.Review.newsCardShownDate.value {
+      daysSinceLastRequest = Calendar.current.dateComponents([.day], from: previousRequest, to: Date()).day ?? 0
+    } else {
+      // First presentation Date, no recorded presentation
+      Preferences.Review.newsCardShownDate.value = Date()
+      return true
+    }
+    
+    if abs(daysSinceLastRequest) < Constants.minDaysBetweenRatingCardPresented {
+      return false
+    }
+    
+    Preferences.Review.newsCardShownDate.value = Date()
```

Comment:
This isn't true though? A card may be generated but the user may not scroll down and see it

Reply:
Thats a great catch, I moved the preference set when the cell is created. Altered at https://github.com/brave/brave-ios/pull/7589/commits/350b2ac81b755972d6afdb631af508545b972454